app/Helpers/database/achievement-creator.php

```diff
@@ -1,70 +1,87 @@
 <?php
 
+use App\Models\Achievement;
+use App\Models\System;
 use App\Models\User;
 use App\Platform\Enums\AchievementFlag;
 
 /**
  * Gets the number of achievements made by the user for each console they have worked on.
  */
-function getUserAchievementsPerConsole(string $username): array
+function getUserAchievementsPerConsole(User $user): array
 {
-    $query = "SELECT COUNT(a.GameID) AS AchievementCount, c.Name AS ConsoleName
-              FROM Achievements as a
-              LEFT JOIN GameData AS gd ON gd.ID = a.GameID
-              LEFT JOIN Console AS c ON c.ID = gd.ConsoleID
-              WHERE a.Author = :author
-              AND a.Flags = :achievementFlag
-              AND gd.ConsoleID NOT IN (100, 101)
-              GROUP BY ConsoleName
-              ORDER BY AchievementCount DESC, ConsoleName";
-
-    return legacyDbFetchAll($query, [
-        'author' => $username,
-        'achievementFlag' => AchievementFlag::OfficialCore,
-    ])->toArray();
+    $userAuthoredAchievements = Achievement::whereHas("game.system", function ($query) {
+        $query->whereNotIn("ID", [System::Hubs, System::Events]);
+      })
+        ->with("game.system")
+        ->where("user_id", $user->id)
+        ->where("Flags", AchievementFlag::OfficialCore)
+        ->get();
+
+    return $userAuthoredAchievements
+        ->groupBy('game.system.Name')
+        ->map(function ($achievements, $systemName) {
+            return [
+                'ConsoleName' => $systemName,
+                'AchievementCount' => $achievements->count(),
+            ];
+        })
+        ->sortByDesc('AchievementCount')
```

Comment:
The left side has a secondary sort on ConsoleName

Reply:
Thanks, I missed that. Resolved in latest.